src/Grid/Column/Transformer/Anonymizer.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This source file is available under the terms of the
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\Grid\Column\Transformer;
+
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\TransformerException;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Column\TransformerInterface;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Util\AdvancedValue;
+
+use function is_string;
+use function md5;
+use function sprintf;
+
+/**
+ * @internal
+ */
+final class Anonymizer implements TransformerInterface
+{
+    public function transform(array $value, array $config): array
+    {
+        if (!isset($config['rule']) || !is_string($config['rule'])) {
+            throw new TransformerException(
+                $this->getName(),
+                sprintf(
+                    'Missing or invalid "rule" configuration (must be a string) for %s transformer.',
+                    $this->getKey()
+                )
+            );
+        }
+
+        $results = [];
+
+        foreach ($value as $val) {
+            $data = $val->getValue();
+            $fieldName = $val->getFieldName() ?? ($config['columnKey'] ?? $this->getKey());
+
+            if (!is_string($data)) {
+                $results[] = new AdvancedValue($val->getType(), $data, $fieldName);
```

Comment:
You can just reuse the $val. Not Necessary to create it new. 

```suggestion
                $results[] = $val;
```